x/ecocredit/server/basket/query_util.go

```diff
@@ -0,0 +1,50 @@
+package basket
+
+import (
+	queryv1beta1 "github.com/cosmos/cosmos-sdk/api/cosmos/base/query/v1beta1"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	gogoproto "github.com/gogo/protobuf/proto"
+	"google.golang.org/protobuf/proto"
+)
+
+func GogoPageReqToPulsarPageReq(from *query.PageRequest) (*queryv1beta1.PageRequest, error) {
+	if from == nil {
+		return &queryv1beta1.PageRequest{}, nil
+	}
+
+	to := &queryv1beta1.PageRequest{}
+	err := GogoToPulsarSlow(from, to)
+	return to, err
+}
+
+func PulsarPageResToGogoPageRes(from *queryv1beta1.PageResponse) (*query.PageResponse, error) {
+	if from == nil {
+		return nil, nil
+	}
```

Comment:
Should we be returning an empty page response here?

Reply:
Well it's not really necessary. This is just a bandaid anyway